json-path/src/main/java/com/jayway/jsonpath/Predicate.java

```diff
@@ -23,6 +24,8 @@
 
     boolean apply(PredicateContext ctx);
 
+    boolean check(TokenStack stack, int idx);
```

Comment:
Appreciated, I manage my own open source project (vjdbc) which is thankfully mature at this point, my day job as an architect (we make a streaming DB) and a side project (www.biovj.com) which is a traditional desktop application for video and projection mapping.  Plus, I'm actively contributing to another 3 opensource projects or so.  So I understand how finding time for JsonPath is hard.  However, JsonPath isn't my project and if the streaming extension requires an architectural change (ie changing/specifying what exactly the Predicate interface is) then I need guidance from you about how to do that change.  So I need you to answer my question.  Should we split the Predicate interface into two things, one that represents logical operators (AND, OR, etc) and one that represents how to evaluate a predicate?  Or should you should be creating facades to support Java8 lambdas instead of overloading the Predicate interface.  For example:
public class NonStreamingPredicateEval {   private Predicate pred;
   public NonStreamingPredicateEval(Predicate pred) { this.pred = pred; }
   public boolean apply(PredicateContext ctx) { return pred.apply(ctx); }}
Either way, its not enough to simply point out the problem. You have to decide upon a solution (or at least a direction) even if you don't personally implement the solution.  What I'm not going to do is implement a solution myself just to have you reject it again later on.  Instead I'm going to wait for you to tell me what you want.
Hunter
PS If lambdas are such an important feature, you should have something in the docs or javadocs that spells out which interfaces are meant to support lambdas so devs don't add methods to those interfaces.  Apologies if you did this already and I just didn't see it.


     On Monday, August 3, 2015 12:02 AM, kallestenflo <notifications@github.com> wrote:
   

 In json-path/src/main/java/com/jayway/jsonpath/Predicate.java:> @@ -23,6 +24,8 @@
>  
>      boolean apply(PredicateContext ctx);
>  
> +    boolean check(TokenStack stack, int idx);
FYI, this is a pet project of mine that I work with on my free time, there is no one else. I appreciate all contributions but I also need to feel comfortable supporting them in the future. I have have made some breaking changes to the API over the years and streaming support is defiantly a feature that I would like. Sorry about the slow pace and lack of comments, I hope to have more time for this project soon. —
Reply to this email directly or view it on GitHub.